types/wordpress__blocks/index.d.ts

```diff
@@ -0,0 +1,403 @@
+// Type definitions for @wordpress/blocks 6.4
+// Project: https://github.com/WordPress/gutenberg/tree/master/packages/blocks/README.md
+// Definitions by: Derek Sifford <https://github.com/dsifford>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.5
+
+import { Dashicon } from '@wordpress/components';
+import { dispatch, select } from '@wordpress/data';
+import { ComponentType } from '@wordpress/element';
+
+export * from './api';
+export { withBlockContentContext } from './block-content-provider';
+
+declare module '@wordpress/data' {
+    function dispatch(key: 'core/blocks'): typeof import('./store/actions');
+    function select(key: 'core/blocks'): typeof import('./store/selectors');
+}
+
+export type BlockAlignment = 'left' | 'center' | 'right' | 'wide' | 'full';
+
+export interface BlockEditProps<T extends Record<string, any>> extends BlockSaveProps<T> {
+    readonly className: string;
+    readonly isSelected: boolean;
+    readonly setAttributes: (attrs: Partial<T>) => void;
+}
+
+export interface BlockIconNormalized {
+    background?: string;
+    foreground?: string;
+    shadowColor?: string;
+    src: Dashicon.Icon | ComponentType;
+}
+
+export type BlockIcon = BlockIconNormalized['src'] | BlockIconNormalized;
+
+export interface BlockSaveProps<T extends Record<string, any>> {
+    readonly attributes: Readonly<T>;
```

Comment:
and whenever using `InnerBlocks` an `innerBlocks` property (array) is also sent to the save function.